src/In.ProjectEKA.HipService/DataFlow/DataFlowClient.cs

```diff
@@ -1,51 +1,125 @@
+    using System.Collections.Generic;
     using System.Linq;
     using System.Net.Http;
+    using System.Net.Mime;
     using System.Text;
-    using System.Threading.Tasks;
     using HipLibrary.Patient;
     using Hl7.Fhir.Serialization;
     using Logger;
+    using Common;
+    using Hl7.Fhir.Model;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Options;
+    using Model;
     using Newtonsoft.Json;
     using Newtonsoft.Json.Serialization;
+    using Task = System.Threading.Tasks.Task;
 
     public class DataFlowClient
     {
         private readonly ICollect collect;
         private readonly HttpClient httpClient;
+        private readonly IServiceScopeFactory serviceScopeFactory;
+        private readonly IOptions<DataFlowConfiguration> dataFlowConfiguration;
+        private readonly IOptions<HipConfiguration> hipConfiguration;
+        private const int MbInBytes = 1000000;
 
-        public DataFlowClient(ICollect collect, HttpClient httpClient)
+        private readonly FhirJsonSerializer serializer = new FhirJsonSerializer(new SerializerSettings());
+
+        public DataFlowClient(ICollect collect,
+            HttpClient httpClient,
+            IServiceScopeFactory serviceScopeFactory,
+            IOptions<DataFlowConfiguration> dataFlowConfiguration,
+            IOptions<HipConfiguration> hipConfiguration)
         {
             this.collect = collect;
             this.httpClient = httpClient;
+            this.serviceScopeFactory = serviceScopeFactory;
+            this.dataFlowConfiguration = dataFlowConfiguration;
+            this.hipConfiguration = hipConfiguration;
         }
 
         public async Task HandleMessagingQueueResult(HipLibrary.Patient.Model.DataRequest dataRequest)
         {
-            (await collect.CollectData(dataRequest))
-                .Map(async entries =>
-                {
-                    var serializer = new FhirJsonSerializer(new SerializerSettings());
-                    var healthRecordEntries = entries.Bundles
-                        .Select(bundle => new Entry(
-                            serializer.SerializeToString(bundle),
-                            "application/json",
-                            "MD5"))
-                        .ToList();
-                    await SendDataToHiu(new DataResponse(dataRequest.TransactionId, healthRecordEntries),
-                        dataRequest.CallBackUrl);
-                    return Task.CompletedTask;
-                });
+            var data = await collect.CollectData(dataRequest);
+            data.Map(async entries => await SendDataToHiu(dataRequest, entries.Bundles.Select(EntryFrom).ToList()));
+        }
+
+        private async Task SendDataToHiu(
+            HipLibrary.Patient.Model.DataRequest dataRequest,
+            IEnumerable<Entry> entries)
+        {
+            await PostTo(dataRequest.CallBackUrl, new DataResponse(dataRequest.TransactionId, entries));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5067092)